src/hermes/commands/deposit/base.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: 2023 Helmholtz-Zentrum Dresden-Rossendorf (HZDR)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileContributor: David Pape
+
+import click
+
+from hermes.model.context import CodeMetaContext
+
+
+# TODO: Abstract base class?
+class BaseDepositPlugin:
+    def __init__(self, click_ctx: click.Context, ctx: CodeMetaContext) -> None:
+        self.click_ctx = click_ctx
+        self.ctx = ctx
+
+    def __call__(self) -> None:
```

Comment:
**thought:** "Explicit is better than implicit" - maybe there should be a method `run` instead?

Reply:
I called it `run` initially but then thought it looked like a step called `run` and wanted to make it more obvious that that's not what it is. Not sure what to call it 🤷🏻‍♂️

Reply:
Well, that's fine I think. The only point where this can get problematic is when we want to call a base implementation in a derived class due to Python attribute mangling with dunder names.